code/game/machinery/outpost_electrolyzer.dm

```diff
@@ -0,0 +1,540 @@
+	if(merit)
+		new /obj/item/merit/bundle(drop_location(), merit)
+		merit = FALSE
+		playsound(src, 'sound/items/poster_being_created.ogg', 10, FALSE)
+		to_chat(user, "<span class='notice'>You retrieve the hydrogen merits.</span>")
+	return ..()
+
+//Hydrogen exchange
+
+/obj/machinery/computer/hydrogen_exchange
+	name = "Hydrogen Exchange"
+	desc = "Credits to Merits at reasonable rates!"
+	icon_screen = "exchange"
+	icon_keyboard = "power_key"
+
+	//GLOB.total_merits_exchanged starts at 0
+	var/merits = NONE
+	var/credits = NONE
+
+/obj/machinery/computer/hydrogen_exchange/attackby(obj/item/I, mob/user)
+	var/value = 0
+	if(istype(I, /obj/item/spacecash/bundle))
+		var/obj/item/spacecash/bundle/C = I
+		value = C.value
+	else if(istype(I, /obj/item/holochip))
+		var/obj/item/holochip/H = I
+		value = H.credits
+	if(value)
+		credits += value
+		to_chat(user, "<span class='notice'>You deposit [I], for a total of [credits] credits.</span>")
+		qdel(I)
+		return
+	if(istype(I, /obj/item/merit/bundle))
+		var/obj/item/merit/bundle/C = I
+		merits += C.value
+		to_chat(user, "<span class='notice'>You deposit [I], for a total of [merits] merits.</span>")
+		qdel(I)
+		return
+	return ..()
+
+/obj/machinery/computer/hydrogen_exchange/proc/meritmultiplier()
+	var/extra = clamp(((GLOB.hydrogen_stored / HYDROGEN_IDEAL) + 1), 0, 2) * 0.3 //results in a number between 0 and .6
+	var/actual = round((0.4 + extra), 0.01) //.4 on low end, 1 on high end
+	return actual
+
+/obj/machinery/computer/hydrogen_exchange/proc/dispense_funds()
+	var/makenoise
+	if(merits)
+		new /obj/item/merit/bundle(drop_location(), merits)
+		merits = 0
+		makenoise = TRUE
+	if(credits)
+		new /obj/item/spacecash/bundle(drop_location(), credits)
+		credits = 0
+		makenoise = TRUE
+	if(makenoise)
+		playsound(src, 'sound/machines/coindrop.ogg', 20, FALSE)
+
+/obj/machinery/computer/hydrogen_exchange/proc/resetmerits() //debug proc
+	GLOB.total_merits_exchanged = 0
+
+/obj/machinery/computer/hydrogen_exchange/proc/convert_to_credits()
+	if(merits)
+		playsound(src, 'sound/machines/pda_button1.ogg', 20, FALSE)
+		var/oldtotal = GLOB.total_merits_exchanged ** MERIT_EXPONENT
+		var/newtotal = (GLOB.total_merits_exchanged + merits) ** MERIT_EXPONENT
+		var/reducedmerits = newtotal - oldtotal
+		GLOB.total_merits_exchanged += merits
+		credits += round(reducedmerits * CREDITS_TO_MERITS, 1)
+		merits = 0
+	else
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 20, FALSE)
+
+/obj/machinery/computer/hydrogen_exchange/proc/convert_to_merits()
+	if(credits)
+		playsound(src, 'sound/machines/pda_button1.ogg', 20, FALSE)
+		merits += round(credits * meritmultiplier() / CREDITS_TO_MERITS, 1)
+		credits = 0
+	else
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 20, FALSE)
+
+/obj/machinery/computer/hydrogen_exchange/AltClick(mob/user)
+	dispense_funds()
+	to_chat(user, "<span class='notice'>You force the credits and merits out of the machine.</span>")
+	return ..()
```

Comment:
same as with the hydrogen pump, this will always open the tile in the top-right
```suggestion
```

Reply:
Unaddressed.